app/controllers/certificates_controller.rb

```diff
@@ -1,30 +1,27 @@
 class CertificatesController < ApplicationController
-  
-  def index
-    @uploads = Certificate.all
-  end
-
-  def upload
+  def new
     @upload = UploadCertificateEvent.new
   end
 
   def create
     @upload = UploadCertificateEvent.create(upload_params)
+
     if @upload.valid?
-      redirect_to certificates_path
+      redirect_to component_path @upload.component_id
     else
       Rails.logger.info(@upload.errors.full_messages)
-      render :upload
+      render :new
     end
   end
 
   private
 
   def upload_params
-    params.require(:certificate).permit(
-      :value,
-      :usage
-    )
-  end
+    @component_id ||= params[:component_id]
```

Comment:
I don't think this needs to be an instance variable

Reply:
👍